tests/test_hsm.py

```diff
@@ -22,3 +22,24 @@ def test_cloudhsm(self):
         tags = client.list_tags(ResourceId=id)
         tag_map = {t["Key"]: t["Value"] for t in tags["TagList"]}
         self.assertTrue("foo" in tag_map)
+
+    def test_cloudhsm_subnet_delete(self):
+        factory = self.replay_flight_data("test_cloudhsm_subnet_delete")
+        client = factory().client("cloudhsmv2")
+        p = self.load_policy(
+            {
+                "name": "cloudhsm",
+                "resource": "cloudhsm-cluster",
+                "filters": [
+                    {"type": "subnet", "key": "SubnetId", "value": "subnet-0544e30702e2e1ad4"},
+                ],
+                "actions": [{"type": "delete"}]
+            },
+            session_factory=factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0].get('ClusterId'), 'cluster-p67ybp4edfr')
```

Comment:
would it make sense to have an assertion here that the subnet in the describe response is as expected?

Reply:
yep, will do!!

Reply:
Addressed!